hermes-management/src/main/java/pl/allegro/tech/hermes/management/api/OfflineRetransmissionEndpoint.java

```diff
@@ -70,22 +71,33 @@ public Response deleteRetransmissionTask(@PathParam("taskId") String taskId) {
     private static class RetransmissionPermissions {
         private final TopicRepository topicRepository;
         private final ManagementRights managementRights;
-
+        private final static Logger logger = LoggerFactory.getLogger(RetransmissionPermissions.class);
 
         private RetransmissionPermissions(TopicRepository topicRepository, ManagementRights managementRights) {
             this.topicRepository = topicRepository;
             this.managementRights = managementRights;
         }
 
         private void ensurePermissionsToBothTopics(OfflineRetransmissionRequest request, ContainerRequestContext requestContext) {
-            Topic sourceTopic = topicRepository.getTopicDetails(TopicName.fromQualifiedName(request.getSourceTopic()));
-            Topic targetTopic = topicRepository.getTopicDetails(TopicName.fromQualifiedName(request.getTargetTopic()));
-            boolean hasPermissions = managementRights.isUserAllowedToManageTopic(sourceTopic, requestContext)
-                    && managementRights.isUserAllowedToManageTopic(targetTopic, requestContext);
+            var targetTopic = topicRepository.getTopicDetails(TopicName.fromQualifiedName(request.getTargetTopic()));
+            var topicValidation = validateSourceTopic(request.getSourceTopic(), requestContext);
+            var userValidation = managementRights.isUserAllowedToManageTopic(targetTopic, requestContext);
+            var hasPermissions = topicValidation
+                    && userValidation;
             if (!hasPermissions) {
-                throw new PermissionDeniedException("User needs permissions to source and target topics.");
+                logger.info("target topic {}", targetTopic);
```

Comment:
Can you add more concise message, just like: `logger.info("User {} has no permissions to make retransmission {}", user, request)`